src/global_scheduler/global_scheduler.cc

```diff
@@ -14,6 +14,45 @@
 #include "state/task_table.h"
 
 /**
+ * Retry the task assignment. If the local scheduler that the task is assigned
+ * to is no longer active, do not retry the assignment.
+ * TODO(rkn): We currently only retry the method if the global scheduler
+ * publishes a task to a local scheduler before the local scheduler has
+ * subscribed to the channel. If we enforce that ordering, we can remove this
+ * retry method.
+ *
+ * @param id The task ID.
+ * @param user_context The global scheduler state.
+ * @param user_data The Task that failed to be assigned.
+ * @return Void.
+ */
+void assign_task_to_local_scheduler_retry(UniqueID id,
+                                          void *user_context,
+                                          void *user_data) {
+  GlobalSchedulerState *state = (GlobalSchedulerState *) user_context;
+  Task *task = (Task *) user_data;
+  CHECK(Task_state(task) == TASK_STATUS_SCHEDULED);
+
+  // If the local scheduler has died since we requested the task assignment, do
+  // not retry again.
+  DBClientID local_scheduler_id = Task_local_scheduler(task);
+  auto it = state->local_schedulers.find(local_scheduler_id);
+  if (it == state->local_schedulers.end()) {
+    return;
+  }
+
+  // The local scheduler is still alive. The failure is most likely due to the
+  // task assignment getting published before the local scheduler subscribed to
+  // the channel. Retry the assignment.
+  auto retryInfo = RetryInfo{
```

Comment:
These `RetryInfo` structs aren't being used anywhere else, so I'm a little concerned about reintroducing them.

I guess the `fail_callback` aspect is very useful, so maybe in a followup PR we should change the methods like `task_table_update` to explicitly take a `fail_callback` and to not take a retry struct.

Reply:
I had the same thought process. I suspect whatever the replacement is will still need something like the `fail_callback`, so I would prefer to just keep it here for now.